PowerUp/app/src/main/java/powerup/systers/com/GameActivity.java

```diff
@@ -19,227 +24,186 @@
-            // TODO Auto-generated catch block
-            e.printStackTrace();
-        }
 
-        String faceImageName = getResources().getString(R.string.face);
-        faceImageName = faceImageName + getmDbHandler().getAvatarFace();
-        try {
-            photoNameField = ourRID.getClass().getField(faceImageName);
-            faceImageView.setImageResource(photoNameField.getInt(ourRID));
-        } catch (NoSuchFieldException | IllegalAccessException
-                | IllegalArgumentException e) {
-            // TODO Auto-generated catch block
-            e.printStackTrace();
-        }
+        setupLayout();
+    }
 
-        String clothImageName = getResources().getString(R.string.cloth);
-        clothImageName = clothImageName + getmDbHandler().getAvatarCloth();
-        try {
-            photoNameField = ourRID.getClass().getField(clothImageName);
-            clothImageView.setImageResource(photoNameField.getInt(ourRID));
-        } catch (NoSuchFieldException | IllegalAccessException
-                | IllegalArgumentException e) {
-            // TODO Auto-generated catch block
-            e.printStackTrace();
-        }
+    private void setupLayout() {
+        mQuestionTextView = (TextView) findViewById(R.id.questionView);
+        mScenarioNameTextView = (TextView) findViewById(R.id.scenarioNameEditText);
+        mAnswers = new ArrayList<>();
+        mGoToMap = (Button) findViewById(R.id.continueButtonGoesToMap);
+        mReplay = (Button) findViewById(R.id.redoButton);
 
-        String hairImageName = getResources().getString(R.string.hair);
-        hairImageName = hairImageName + getmDbHandler().getAvatarHair();
-        try {
-            photoNameField = ourRID.getClass().getField(hairImageName);
-            hairImageView.setImageResource(photoNameField.getInt(ourRID));
-        } catch (NoSuchFieldException | IllegalAccessException
-                | IllegalArgumentException e) {
-            // TODO Auto-generated catch block
-            e.printStackTrace();
-        }
+        UiUtils.setupEyesFaceClothesHair(this, mDbHandler);
 
-        // Update Scene
+        // Update scene
         updateScenario();
-        if (scene.getReplayed() == 1) {
-            goToMap.setAlpha((float) 0.0);
-            replay.setAlpha((float) 0.0);
+        if (mScene.getReplayed() == 1) {
+            mGoToMap.setAlpha((float) 0.0);
+            mReplay.setAlpha((float) 0.0);
         }
-        // Set the ArrayAdapter as the ListView's adapter.
-        mainListView.setAdapter(listAdapter);
-        mainListView
-                .setOnItemClickListener(new AdapterView.OnItemClickListener() {
+
+        setupListView();
+
+        UiUtils.setupProgressBars(this, mDbHandler);
+    }
+
+    private void setupListView() {
+        mListAdapter = new ArrayAdapter<>(this, R.layout.simplerow, new ArrayList<String>());
+
+        ListView mainListView = (ListView) findViewById(R.id.mainListView);
```

Comment:
Move lines 79-81 in setupLayout() so that mListAdapter is initialized before calling updateScenario() else it will give runtime error of mListadapter being a null object on which various functions are being called in updateQA()